crypto_track/blog.md

```diff
@@ -0,0 +1,574 @@
+    const { requestId } = await CourierClient.send({
+        message: {
+            to: {
+                email: email,
+            },
+            channels: {
+                "email": {
+                    override: {
+                        subject: `New update on the ${metric} of ${coin}`
+                    }
+                }
+            },
+            template: "NS6KJEXFZNMYVCHZD8AQ1X7NAVQR",
+            data: {
+                metric,
+                targetValue,
+                latestValue,
+                change,
+                coin
+            },
+        },
+    });
+    return requestId;
+}
+
+
+```
+
+> One thing to note here is, the `template` field should match the Notification template ID we created earlier, which you can find on the details page. 
+
+
+Now that we have defined all our notification logic, we will create our express app on the `index.js` file:
+
+
+```js
+
+import express from "express";
+import cors from "cors";
+import SupabaseClient from "./supabase.js";
+import notify from "./notification.js";
+
+const app = express();
+
+app.use(cors());
+
+app.post("/", async (req, res) => {
+  res.set("Content-Type", "application/json");
+  try {
+    const emails = await notify(SupabaseClient);
+    return res.send(JSON.stringify({ message: "success", emails }));
+  }
+  catch (error) {
+    console.log({ error })
+    return res.status(500).send(JSON.stringify(error));
+  }
+});
+
+const PORT = process.env.PORT || 5000
+app.listen(PORT, () => {
+  console.log(`Running on port ${PORT}.`);
+});
+
+export default app;
+
+```
+
+Sending a post request to the root route will call the `notify` method to send emails to  subscribes if there is a desired change in the market.
+
+
+#### Postgers CRON Server
+
+Connecting our webhook to our Postgres cron server is the last step. Two extensions must first be added to our Postgres database. Run the following scripts to add the "http" and "pg cron" extensions in the SQL editor on the Supabase project dashboard:
+
+```pgsql
+
+create extension if not exists http;
+create extension if not exists pg_cron;
+
+``` 
+After completing that, we can configure our webhook to be called periodically. But in order for the webhook to be accessible by our Postgres cron server, we either need to deploy it or use a service like [ngrok](https://ngrok.com). Then we can run the script below to set up a cron job that runs every minute:
+
+
+```pgsql
+
+select
+  cron.schedule(
+    'market-data', 
+    '* * * * *',
+    $$
+    select status
+    from
+       http_post(
+        'https://my-crypto-track-webhook.url/',
+        '{}', -- payload
+        'application/json'
+      )
+    $$
+  );
+ 
+
+```
+
+## Conclusions
+
+And we are done, I hope you enjoyed this tutorial. You can find the full source code on [Github](https://github.com/eyuelberga/CryptoTrack/) and also check about the demo [here](https://eyuelberga.github.io/CryptoTrack/).
```

Comment:
What are some challenges you faced while building this? How would you suggest the readers challenge themselves to take this project to the next level? What was your experience like using Courier?